fsh/ig-data/input/pagecontent/index.md

```diff
@@ -67,161 +94,280 @@ An example call from an app to the EHR client is presented below.
+where EHR should go to after the ui message has been handled. An activity may
+specify additional parameters that can be included in the call as additional
+properties.
 
-The `ui.done` messageType instructs the EHR to close the activity hosting the SMART app, and optionally navigates the user to an alternate activity.
+The `activityParameters` property conveys parameters specific to an activity
+type. See the SMART Web Messaging [Activity Catalog] for details.
 
 *Note:* A SMART app launched in the context of CDS Hooks should generally not
-need to specify an `activityType` or `activityParameters` with the `ui.done` message, because the EHR
-tracks the context in which the app was launched (e.g., order entry) and can
-navigate to the appropriate follow-up screen based on this context.
+need to specify an `activityType` or `activityParameters` with the `ui.done`
+message, because the EHR tracks the context in which the app was launched (e.g.,
+order entry) and can navigate to an appropriate follow-up screen based on this
+context.
 
-An example of a `ui.done` message is shown below:
+An example of a `ui.done` message from an app to the EHR is shown below:
 
 ```js
-appWindow.postMessage({
+targetWindow.postMessage({
   "messageId": "<some new guid>",
   "messageType": "ui.done",
   "payload": {
     "activityType": "problem-add",
     "activityParameters": {
-      // Each ui activity defines its optional+required params
+      // Each ui activity defines its optional and required params.  See the
+      // Activity Catalog for more details.
       "problem": {
         "resourceType": "Condition",
         "patient": "123",
       }
     }
   }
-}, targetOrigin)
-
+}, targetOrigin);
 ```
 
-Similarly, the SMART app can use the `ui.LaunchActivity` message type to request navigation to an alternate activity without closing the app:
+Similarly, the SMART app can use the `ui.LaunchActivity` message type to request
+navigation to a different activity *without* closing the app:
 
 ```js
-
-
-appWindow.postMessage({
+targetWindow.postMessage({
   "messageId": "<some new guid>",
   "messageType": "ui.launchActivity",
   "payload": {
     "activityType": "problem-add",
     "activityParameters": {
-      // Each ui activity defines its optional+required params
+      // Each ui activity defines its optional and required params.  See the
+      // Activity Catalog for more details.
       "problem": {
         "resourceType": "Condition",
         "patient": "123",
       }
     }
   }
-}, targetOrigin)
+}, targetOrigin);
 ```
 
-The EHR SHALL respond to all `ui` message types with a payload that includes a boolean `success` parameter and an optional `details` string:
+The EHR SHALL respond to all `ui` message types with a payload that includes a
+boolean `success` parameter and an optional `details` string:
+
 ```js
-{
-  "success": true | false,
-  "details": "string explanation for user (optional)"
-}
+event.source.postMessage({
```

Comment:
This is missing the listener code from the examples above -- so `event` isn't clearly defines, and it's not clear whether event.origin has been checked and is safe to postMessage to).

Reply:
Fair enough.  I thought about adding the event listener code back in, but it muddled the example.